libs/git/clone.go

```diff
@@ -27,11 +27,14 @@ type cloneOptions struct {
 	TargetPath string
 }
 
-func (opts cloneOptions) args() []string {
-	args := []string{"clone", opts.RepositoryUrl, opts.TargetPath, "--depth=1", "--no-tags"}
+func (opts cloneOptions) args(shallow bool) []string {
```

Comment:
Please add this as a field in opts. It's not descriptive to have to call `.opts(true)`.

Reply:
Good point. I kept it separate because its behavioural rather than information about the destination, but this better.